octobot_trading/modes/abstract_trading_mode.py

```diff
@@ -354,3 +556,213 @@ async def edit_order(self, order,
                 updated_current_price=edited_current_price,
             )
         return changed
+
+    async def _wait_for_run_data_init(self, timeout) -> bool:
+        if self.exchange_manager.is_backtesting:
+            raise NotImplementedError("Not implemented in backtesting")
+        try:
+            await asyncio.wait_for(
+                commons_tree.EventProvider.instance().get_or_create_event(
+                    self.exchange_manager.bot_id,
+                    commons_tree.get_exchange_path(
+                        self.exchange_manager.exchange_name,
+                        common_enums.InitializationEventExchangeTopics.CANDLES.value
+                    ),
+                    allow_creation=False
+                ).wait(),
+                timeout
+            )
+            if not self.exchange_manager.is_future:
+                return True
```

Comment:
I don't think this is mandatory because the path will not exists when not in future